### DISCUSSION TITLE
Missing sbt dependencies from repo1.maven.org

### DISCUSSION THREAD

**Discussion by gervaisb** [sbt/sbt#7205] [Q&A] [answered]
Hi there, 

As suggest by @SethTisue (https://github.com/sbt/sbt/issues/7202#issuecomment-1500262265) I open a discussion on that subject.

On Apr 7, 2023 10:20am GMT+0200 (The issue may have started sooner) we noticed that our CI was unable to download artifacts from https://repo1.maven.org/maven2/com/typesafe/sbt/  many of them are "officials" sbt plugins and it seems that they disaperaded from the maven repo.

+ https://repo1.maven.org/maven2/com/typesafe/sbt/sbt-native-packager_2.12_1.0/1.7.6/sbt-native-packager-1.7.6.pom
+ https://repo1.maven.org/maven2/com/typesafe/sbt/sbt-js-engine_2.12_1.0/1.2.3/sbt-js-engine-1.2.3.pom
+ https://repo1.maven.org/maven2/com/lightbend/sbt/sbt-javaagent_2.12_1.0/0.1.6/sbt-javaagent-0.1.6.pom
+ https://repo1.maven.org/maven2/com/github/cb372/sbt-explicit-dependencies_2.12_1.0/0.2.16/sbt-explicit-dependencies-0.2.16.pom
+ https://repo1.maven.org/maven2/com/typesafe/sbt/sbt-twirl_2.12_1.0/1.5.1/sbt-twirl-1.5.1.pom
+ https://repo1.maven.org/maven2/com/typesafe/sbt/sbt-web_2.12_1.0/1.4.4/sbt-web-1.4.4.pom
+ ...

Here is a sample error message:
> [error] Error downloading com.typesafe.sbt:sbt-web;sbtVersion=1.0;scalaVersion=2.12:1.4.4
[error]   Not found
[error]   Not found
[error]   not found: https://repo1.maven.org/maven2/com/typesafe/sbt/sbt-web_2.12_1.0/1.4.4/sbt-web-1.4.4.pom
[error]   not found: /builds/acme/app/.ivy2/localcom.typesafe.sbt/sbt-web/scala_2.12/sbt_1.0/1.4.4/ivys/ivy.xml
[error]   checksum format error: /builds/acme/app/.coursier/cache/https/repo.scala-sbt.org/scalasbt/sbt-plugin-releases/com.typesafe.sbt/sbt-web/scala_2.12/sbt_1.0/1.4.4/ivys/ivy.xml.sha1
[error]   checksum format error: /builds/acme/app/.coursier/cache/https/repo.typesafe.com/typesafe/ivy-releases/com.typesafe.sbt/sbt-web/scala_2.12/sbt_1.0/1.4.4/ivys/ivy.xml.sha1

Has anybody encountered the same kind of errors ? 
Do you have any idea on what can be the issue and how can we solve it ?

Thanks

**Comment by SethTisue** [comment]
It seems like this must somehow be related to #7202, but I cannot understand, whether due to ignorance or lack of imagination, what the connection could be 😕

Is it possible that sbt is only even looking in these places at all, because it failed to find something that it should have found on repo.scala-sbt.org?

**Reply by gervaisb** [reply]
I tried on my machine with `SBT_OPTS: "-Dsbt.coursier.home=.coursier -ivy .ivy2 -sbt-dir .sbt"` (in the hope of working on a fresh workspace) and I got exactly the same errors.

Adding `-debug` and `-verbose` does not give more than a quite useless stack trace:

> [error] sbt.librarymanagement.ResolveException: Error downloading com.typesafe.sbt:sbt-native-packager;sbtVersion=1.0;scalaVersion=2.12:1.7.6
[error] 	at lmcoursier.CoursierDependencyResolution.unresolvedWarningOrThrow(CoursierDependencyResolution.scala:344)
[error] 	at lmcoursier.CoursierDependencyResolution.$anonfun$update$38(CoursierDependencyResolution.scala:313)
[error] 	at scala.util.Either$LeftProjection.map(Either.scala:573)
[error] 	at lmcoursier.CoursierDependencyResolution.update(CoursierDependencyResolution.scala:313)
[error] 	at sbt.librarymanagement.DependencyResolution.update(DependencyResolution.scala:60)
[error] 	at sbt.internal.LibraryManagement$.resolve$1(LibraryManagement.scala:59)
[error] 	at sbt.internal.LibraryManagement$.$anonfun$cachedUpdate$12(LibraryManagement.scala:133)
[error] 	at sbt.util.Tracked$.$anonfun$lastOutput$1(Tracked.scala:73)
[error] 	at sbt.internal.LibraryManagement$.$anonfun$cachedUpdate$20(LibraryManagement.scala:146)
[error] 	at scala.util.control.Exception$Catch.apply(Exception.scala:228)
[error] 	at sbt.internal.LibraryManagement$.$anonfun$cachedUpdate$11(LibraryManagement.scala:146)
[error] 	at sbt.internal.LibraryManagement$.$anonfun$cachedUpdate$11$adapted(LibraryManagement.scala:127)
[error] 	at sbt.util.Tracked$.$anonfun$inputChangedW$1(Tracked.scala:219)
[error] 	at sbt.internal.LibraryManagement$.cachedUpdate(LibraryManagement.scala:160)
[error] 	at sbt.Classpaths$.$anonfun$updateTask0$1(Defaults.scala:3687)
[error] 	at scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error] 	at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:62)
[error] 	at sbt.std.Transform$$anon$4.work(Transform.scala:68)
[error] 	at sbt.Execute.$anonfun$submit$2(Execute.scala:282)
[error] 	at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:23)
[error] 	at sbt.Execute.work(Execute.scala:291)
[error] 	at sbt.Execute.$anonfun$submit$1(Execute.scala:282)
[error] 	at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:265)
[error] 	at sbt.CompletionService$$anon$2.call(CompletionService.scala:64)
[error] 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error] 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
[error] 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error] 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[error] 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[error] 	at java.base/java.lang.Thread.run(Thread.java:829)

**Comment by mkurz** [comment]
I am pretty sure these dependencies (at least the posted versions) never existed on maven central. It's only now that repo.scala-sbt.org is down that SBT falls back and tries to retrieve them from maven central.

Pretty sure this will just work again when the repo is up again

**Reply by gheine** [reply]
Was just gonna say the same ^^. Since the dependencies fail to download from repo.scala-sbt.org, sbt will try maven central, which also fails because they were never there in the first place.

**Reply by gervaisb** [reply]
I hope you are right 🤞  

I have no idea if they were on maven central. I have access to the logs of the previous (successful) builds. Unfortunately, the "source" of those dependencies is not shown.

**Reply by jlouns** [reply]
FWIW, on my local machine, `~/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/com/lightbend/sbt/sbt-javaagent_2.12_1.0/0.1.6` is empty, and `~/Library/Caches/Coursier/v1/https/repo.scala-sbt.org/scalasbt/sbt-plugin-releases/com.lightbend.sbt/sbt-javaagent/scala_2.12/sbt_1.0/0.1.6` has artifacts in it, which would support this idea.

**Comment by SethTisue** [comment]
maybe @mkurz and I are right, but if people want to consider other possible explanations, maybe there are clues buried somewhere in https://github.com/sbt/sbt/issues/6294 — I'm out of time today to go digging through it

**Comment by gervaisb** [comment, marked as answer]
As expected. This issue has been solved with #7202   

Thanks for all.
